src/include/ITable.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <cstdlib>
+#include <vector>
+
+#include "Filters.h"
+
+namespace Tables::Basic {
```

Comment:
Would call it `Tables::Scalar` as scalar is the opposite to vectorised calculations 🧐